[CI] DataNodeRequestSenderTests testLimitConcurrentNodes failing

**Build Scans:**
- [elasticsearch-periodic #9888 / openjdk21_checkpart3_java-fips-matrix](https://gradle-enterprise.elastic.co/s/cwnbjujfueeic)
- [elasticsearch-pull-request #84636 / part-3](https://gradle-enterprise.elastic.co/s/vdbmma5somlju)
- [elasticsearch-pull-request #84447 / part-3](https://gradle-enterprise.elastic.co/s/faqibp5owyod4)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:test" --tests "org.elasticsearch.xpack.esql.plugin.DataNodeRequestSenderTests.testLimitConcurrentNodes" -Dtests.seed=729ED1C118D54B33 -Dtests.locale=su-ID -Dtests.timezone=Africa/Harare -Druntime.java=21 -Dtests.fips.enabled=true
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.plugin.DataNodeRequestSenderTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testLimitConcurrentNodes'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: <2>
     but: was <1>
```

**Issue Reasons:**
- [main] 3 failures in test testLimitConcurrentNodes (0.7% fail rate in 426 executions)
- [main] 2 failures in step part-3 (1.1% fail rate in 183 executions)
- [main] 2 failures in pipeline elasticsearch-pull-request (1.1% fail rate in 183 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -521,9 +521,6 @@ tests:
 - class: org.elasticsearch.xpack.esql.qa.mixed.MixedClusterEsqlSpecIT
   method: test {csv-spec:lookup-join.MvJoinKeyFromRowExpanded}
   issue: https://github.com/elastic/elasticsearch/issues/132604
-- class: org.elasticsearch.xpack.esql.plugin.DataNodeRequestSenderTests
-  method: testLimitConcurrentNodes
-  issue: https://github.com/elastic/elasticsearch/issues/132607
 - class: org.elasticsearch.xpack.test.rest.XPackRestIT
   method: test {p0=esql/60_usage/Basic ESQL usage output (telemetry) non-snapshot version}
   issue: https://github.com/elastic/elasticsearch/issues/132608

--- a/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/plugin/DataNodeRequestSenderTests.java
+++ b/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/plugin/DataNodeRequestSenderTests.java
@@ -306,13 +306,16 @@ public class DataNodeRequestSenderTests extends ComputeTestCase {
     }
 
     public void testLimitConcurrentNodes() {
-        var targetShards = List.of(
-            targetShard(shard1, node1),
-            targetShard(shard2, node2),
-            targetShard(shard3, node3),
-            targetShard(shard4, node4),
-            targetShard(shard5, node5)
-        );
+        final int shards = 10;
+        var targetShards = new ArrayList<DataNodeRequestSender.TargetShard>(shards);
+        for (int i = 0; i < shards; i++) {
+            targetShards.add(
+                targetShard(
+                    new ShardId("index", "n/a", i),
+                    DiscoveryNodeUtils.builder("node-" + i).roles(Set.of(DATA_HOT_NODE_ROLE)).build()
+                )
+            );
+        }
 
         var concurrency = randomIntBetween(1, 2);
         AtomicInteger maxConcurrentRequests = new AtomicInteger(0);
@@ -335,10 +338,10 @@ public class DataNodeRequestSenderTests extends ComputeTestCase {
                 listener.onResponse(new DataNodeComputeResponse(DriverCompletionInfo.EMPTY, Map.of()));
             });
         }));
-        assertThat(sent.size(), equalTo(5));
+        assertThat(sent.size(), equalTo(shards));
         assertThat(maxConcurrentRequests.get(), equalTo(concurrency));
-        assertThat(response.totalShards, equalTo(5));
-        assertThat(response.successfulShards, equalTo(5));
+        assertThat(response.totalShards, equalTo(shards));
+        assertThat(response.successfulShards, equalTo(shards));
         assertThat(response.failedShards, equalTo(0));
     }
 

